pandas/core/arrays/categorical.py

```diff
@@ -89,18 +89,25 @@ def f(self, other):
             else:
                 other_codes = other._codes
 
-            na_mask = (self._codes == -1) | (other_codes == -1)
+            mask = (self._codes == -1) | (other_codes == -1)
             f = getattr(self._codes, op)
             ret = f(other_codes)
-            if na_mask.any():
+            if mask.any():
                 # In other series, the leads to False, so do that here too
-                ret[na_mask] = False
+                ret[mask] = False
             return ret
 
         if is_scalar(other):
             if other in self.categories:
                 i = self.categories.get_loc(other)
-                return getattr(self._codes, op)(i)
+                ret = getattr(self._codes, op)(i)
+
+                # check for NaN in self
+                mask = (self._codes == -1)
+                if mask.any():
```

Comment:
Why do we need `mask.any()`? Can we not just have `ret[mask] = False`

Reply:
Hi Tom, it seems the CI config is buggy. I checked several PR have similar errors. Need help~

Reply:
@yanglinlee can you comment on https://github.com/pandas-dev/pandas/pull/26514#discussion_r289191338?

Reply:
I think it is a good idea. Let's make it simpler and quicker.